### PR TITLE
RTDEV-63005 - Support generate key-pair using jfrog cli evd.

### PR DESCRIPTION
**Pull request by osaidwtd**
- [ ] All [tests](https://github.com/jfrog/jfrog-client-go#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [ ] All [static analysis checks](https://github.com/jfrog/jfrog-client-go/actions/workflows/analysis.yml) passed.
- [ ] This pull request is on the master branch.
- [ ] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

File: artifactory/services/trustedkeys.go [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+func (tks *TrustedKeysService) UploadTrustedKey(params TrustedKeyParams) (*TrustedKeyResponse, error) {
+	if params.Alias == "" {
+		return nil, errorutils.CheckErrorf("key alias cannot be empty")
+	}
+	if params.PublicKey == "" {
+		return nil, errorutils.CheckErrorf("public key cannot be empty")
+	}
+
+	// Check if alias already exists
+	exists, err := tks.CheckAliasExists(params.Alias)
```
Comment (selalererjfrog) [inline artifactory/services/trustedkeys.go]: Why do we check?
It is not synchronized in any case, so by the time you'll get to SendPost it might have been created

**Review by selalererjfrog — APPROVED** [review]

**Review by mnsboev — COMMENTED** [review]
Please add new methods of artifactory service manager to readme

File: artifactory/services/trustedkeys_test.go [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+			name:          "404 with empty response",
+			statusCode:    404,
+			responseBody:  "",
+			expectedError: "trusted keys API returned status 404: 404 page not found - trusted keys API endpoint not available",
+		},
+		{
+			name:          "401 with empty response",
+			statusCode:    401,
+			responseBody:  "",
+			expectedError: "trusted keys API returned status 401: Unauthorized - invalid or expired authentication token",
```
Comment (RemiBou) [inline artifactory/services/trustedkeys_test.go:179]: are all those test needed ? seems redundant ( and generated )

**Review by RemiBou — APPROVED** [review]

File: artifactory/services/trustedkeys_test.go (hunk 2 above)
Comment (osaidwtd) [reply artifactory/services/trustedkeys_test.go:179]: @RemiBou AI response to you:
The tests are valuable and should be kept because:
     1.  Complete coverage - Tests both happy and edge cases
     2. Production scenarios - Reflects real server behaviors
     3. Error message quality - Ensures users get helpful messages even when servers don't
     4. Regression prevention - Protects against future code changes

Comment (RemiBou) [reply artifactory/services/trustedkeys_test.go:179]: why no test all the http response code then ? dont forget this code will be read by human
